examples/spark2/src/main/scala/org/apache/carbondata/examples/ConcurrencyTest.scala

```diff
@@ -0,0 +1,355 @@
+      results.get(i).get()
+    }
+    for (i <- 0 until results.size()) {
+      timeArray(i) = results.get(i).get().time
+    }
+    val sortTimeArray = timeArray.sorted
+
+    val time90 = ((sortTimeArray.length) * 0.9).toInt - 1
+    val time99 = ((sortTimeArray.length) * 0.99).toInt - 1
+    print("90%:" + sortTimeArray(time90) + " s," +
+      "\t99%:" + sortTimeArray(time99) + " s," +
+      "\tlast:" + sortTimeArray.last + " s," +
+      "\t" + sql +
+      "\taverage:" + (timeArray.sum / timeArray.length) + " s," +
+      "\t" + sqlResult.mkString(",") + "\t")
+  }
+
+  case class Results(time: Double, sqlResult: Array[Row])
+
+
+  class QueryTask(spark: SparkSession, query: String)
+    extends Callable[Results] with Serializable {
+    override def call(): Results = {
+      var result: Array[Row] = null
+      val rt = time {
+        result = spark.sql(query).head(1)
+      }
+      if (ResultIsEmpty) {
+        Results(rt, Array.empty[Row])
+      } else {
+        Results(rt, result)
+      }
+    }
+  }
+
+  // run testcases and print comparison result
+  def runTest(spark: SparkSession, table1: String, table2: String): Unit = {
+    // run queries on parquet and carbon
+    runQueries(spark, table1)
+    // do GC and sleep for some time before running next table
+    System.gc()
+    Thread.sleep(1000)
+    System.gc()
+    Thread.sleep(1000)
+    runQueries(spark, table2)
+  }
+
+  def time(code: => Unit): Double = {
+    val start = System.currentTimeMillis()
+    code
+    // return time in second
+    (System.currentTimeMillis() - start).toDouble / 1000
+  }
+
+  def doParameter(arr: Array[String]): Unit = {
+    if (arr.length > 0) {
+      totalNum = arr(0).toInt
+    }
+    if (arr.length > 1) {
+      ThreadNum = arr(1).toInt
+    }
+    if (arr.length > 2) {
+      TaskNum = arr(2).toInt
+    }
+    if (arr.length > 3) {
+      ResultIsEmpty = if (arr(3).equalsIgnoreCase("true")) {
+        true
+      } else if (arr(3).equalsIgnoreCase("false")) {
+        true
+      } else {
+        throw new Exception("error parameter, should be true or false")
+      }
+    }
+  }
+
+  def main(args: Array[String]): Unit = {
+    CarbonProperties.getInstance()
+      .addProperty("carbon.enable.vector.reader", "true")
+      .addProperty("enable.unsafe.sort", "true")
+      .addProperty("carbon.blockletgroup.size.in.mb", "32")
+      .addProperty(CarbonCommonConstants.ENABLE_UNSAFE_COLUMN_PAGE_LOADING, "true")
+    import org.apache.spark.sql.CarbonSession._
+    val rootPath = new File(this.getClass.getResource("/").getPath
+      + "../../../..").getCanonicalPath
+    val storeLocation = s"$rootPath/examples/spark2/target/store"
+
+    val spark = SparkSession
+      .builder()
+      .master("local[8]")
+      .enableHiveSupport()
+      .config("spark.driver.host", "127.0.0.1")
+      .getOrCreateCarbonSession(storeLocation)
+    spark.sparkContext.setLogLevel("warn")
+
+    doParameter(args)
```

Comment:
Change the method name to initParameters

Reply:
ok, done